modin/core/dataframe/pandas/dataframe/utils.py

```diff
@@ -0,0 +1,266 @@
+    key: Optional[Callable] = None,
+) -> np.ndarray:
+    """
+    Determine quantiles from the given samples.
+
+    This function takes as input the quantiles calculated over all partitions from
+    `sample_func` defined above, and determines a final NPartitions.get() quantiles
+    to use to roughly sort the entire dataframe. It does so by collating all the samples
+    and computing NPartitions.get() quantiles for the overall set.
+
+    Parameters
+    ----------
+    df : PandasDataframe
+        The Modin Dataframe calling this function.
+    samples : np.ndarray
+        The samples computed by ``get_partition_quantiles_for_sort``.
+    method : str, default: linear
+        The method to use when picking quantiles.
+    key : Callable, default: None
+        The key to use on the samples when picking pivots.
+
+    Returns
+    -------
+    np.ndarray
+        A list of overall quantiles.
+    """
+    # We don't call `np.unique` on the samples, since if a quantile shows up in multiple
+    # partition's samples, this is probably an indicator of skew in the dataset, and we
+    # want our final partitions to take this into account.
+    # We need to use numpy to concatenate the samples since the sample from each partition is
+    # a NumPy array, and we want one flattened array of samples.
+    all_pivots = np.concatenate(samples).flatten()
+    if key is not None:
+        all_pivots = key(all_pivots)
+    # We don't want to pick very many quantiles if we have a very small dataframe.
+    num_quantiles = len(df._partitions)
+    quantiles = [i / num_quantiles for i in range(1, num_quantiles)]
+    overall_quantiles = _find_quantiles(all_pivots, quantiles, method)
+    return overall_quantiles
+
+
+def split_partitions_using_pivots_for_sort(
+    modin_frame: PandasDataframe,
+    df: pandas.DataFrame,
+    column: str,
+    pivots: np.ndarray,
+    ascending: bool,
+    **kwargs: dict,
+) -> "tuple[pandas.DataFrame]":
+    """
+    Split the given dataframe into the partitions specified by `pivots`.
+
+    This function takes as input a row-axis partition, as well as the quantiles determined
+    by the `pivot_func` defined above. It then splits the input dataframe into NPartitions.get()
+    dataframes, with the elements in the i-th split belonging to the i-th partition, as determined
+    by the quantiles we're using.
+
+    Parameters
+    ----------
+    modin_frame : PandasDataframe
+        The Modin Dataframe calling this function.
+    df : pandas.Dataframe
+        The partition to split.
+    column : str
+        The major column to sort by.
+    pivots : np.ndarray
+        The quantiles to use to split the data.
+    ascending : bool
+        The ascending flag.
+    **kwargs : dict
+        Additional keyword arguments.
+
+    Returns
+    -------
+    tuple[pandas.DataFrame]
+        A tuple of the splits from this partition.
+    """
+    if not ascending and modin_frame.dtypes[column] != object:
```

Comment:
this condition looks weird... why are we ignoring `ascending` if sorting an object column? what if we were given a `key()` function?..

Reply:
We can ignore `key` here since the pivots will already have `key` applied to them in the `pick_pivots_from_sample` function.

This code is just to make it so that we split our data correctly if `ascending=False`. If we are dealing with a numeric dtype, I can pass in a reverse sorted list of pivots to `np.digitize`, and it will work. If we are dealing with an object type, say `str`, passing in a reverse sorted list will break `np.searchsorted`:
```python
In [35]: np.searchsorted(["cat", "at"], "bat")
Out[35]: 2

In [36]: sorted(["bat", "cat", "at"])
Out[36]: ['at', 'bat', 'cat']

In [37]: sorted(["bat", "cat", "at"])[::-1]
Out[37]: ['cat', 'bat', 'at']
```
So in order to handle the case when `ascending=False` on a numeric dtype, I simply reverse the pivots. To handle the `ascending=False` case on an object dtype, I need to get the new partition indices as if `ascending=True`, and then reverse them (hence the `groupby_col = len(pivots) - groupby_col` below).

I can add a comment to explain this workflow!

Reply:
yes, a comment would be awesome!